xmcl-runtime/modpack/ModpackService.ts

```diff
@@ -102,18 +102,12 @@ export class ModpackService extends AbstractService implements IModpackService {
 
     let name = instance.name
     let idx = 1
+    let candidatePath = this.getPath('instances', name)
 
-    while (true) {
-      try {
-        await mkdir(this.getPath('instances', name))
-        break
-      } catch (e) {
-        if (isSystemError(e) && e.code === 'EEXIST') {
-          name = `${name}-${idx++}`
-          continue
-        }
-        throw e
-      }
+    // Find an available directory name without creating the directory
+    while (existsSync(candidatePath)) {
```

Comment:
Actually, we can totally remove this logic. The folder detection is done in createInstance method.